File: my-folder/0392-is-subsequence/solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int s_index = 0;
        for (auto c: t) {
            if (c == s[s_index]) s_index++;
        }
        // if s is subseq of t s_index will be equal to s.size()
        return s_index == s.size();
    }
};

File: my-folder/0392-is-subsequence/solution.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        let mut s = s.chars().peekable();
        for c in t.chars() {
            match s.peek() {
                Some(&ch) if ch == c => {
                    s.next();
                },
                _ => {},
            }
        }
        s.next().is_none()
    }
}
